haskell/test/Test/Ligo/BaseDAO/Common.hs

```diff
@@ -271,11 +274,30 @@ originateLigoDao =
 createSampleProposal
   :: (MonadNettest caps base m, HasCallStack)
   => Int -> Address -> TAddress Parameter -> m ProposalKey
-createSampleProposal counter dodOwner1 dao = do
+createSampleProposal counter dodOwner dao = do
+  let (pk, action) = createSampleProposal_ counter dodOwner dao
+  withSender dodOwner $ inBatch action
+  pure pk
+
+createSampleProposal_
+  :: (MonadOps m, HasCallStack)
+  => Int -> Address -> TAddress Parameter -> (ProposalKey, m ())
+createSampleProposal_ counter dodOwner1 dao =
   let params = ProposeParams
         { ppFrozenToken = 10
         , ppProposalMetadata = lPackValueRaw @Integer $ fromIntegral counter
         }
+  in (makeProposalKey params dodOwner1, call dao (Call @"Propose") params)
 
-  withSender dodOwner1 $ call dao (Call @"Propose") params
-  pure $ (makeProposalKey params dodOwner1)
+createSampleProposals
```

Comment:
Can you please add this TODO, so we can reconsider it later if we have more use for it?
```suggestion
-- TODO consider making this polymorphic on the input/output size
createSampleProposals
```